workflow/executor_unit_test.go

```diff
@@ -0,0 +1,436 @@
+package workflow //nolint:testpackage // Tests private members in the packages
+
+import (
+	"go.arcalot.io/assert"
+	"go.flow.arcalot.io/engine/internal/step"
+	"go.flow.arcalot.io/pluginsdk/schema"
+	"testing"
+)
+
+var testLifecycleStage = step.LifecycleStageWithSchema{
+	LifecycleStage: step.LifecycleStage{},
+	InputSchema: map[string]*schema.PropertySchema{
+		"scopeA": schema.NewPropertySchema(
+			schema.NewScopeSchema(schema.NewObjectSchema("testObjectA", map[string]*schema.PropertySchema{})),
+			nil,
+			true,
+			nil,
+			nil,
+			nil,
+			nil,
+			nil,
+		),
+		"listA": schema.NewPropertySchema(
+			schema.NewListSchema(
+				schema.NewScopeSchema(
+					schema.NewObjectSchema("testObjectB", map[string]*schema.PropertySchema{}),
+				),
+				nil,
+				nil,
+			),
+			nil,
+			true,
+			nil,
+			nil,
+			nil,
+			nil,
+			nil,
+		),
+		"intA": schema.NewPropertySchema(
+			schema.NewIntSchema(nil, nil, nil),
+			nil,
+			true,
+			nil,
+			nil,
+			nil,
+			nil,
+			nil,
+		),
+	},
+	Outputs: map[string]*schema.StepOutputSchema{
+		"outputC": schema.NewStepOutputSchema(
```

Comment:
Just out of curiosity, why is this schema named `"outputC"` instead of "outputA"?  If it's because it's trying to match the `"testObjectC"`, then why is the schema at line 23 not named "listB"?